service/worker/scanner/scanner_test.go

```diff
@@ -40,9 +64,249 @@ func TestScannerSuite(t *testing.T) {
 
 func (s *scannerTestSuite) SetupTest() {
 	s.mockCtrl = gomock.NewController(s.T())
+	s.mockResource = resource.NewTest(s.T(), s.mockCtrl, metrics.Worker)
+	s.mockScope = tally.NoopScope
+	s.mockWorker = worker2.NewMockWorker(s.mockCtrl)
+	s.mockClient = new(mocks.Client)
 
+	s.scanner = &Scanner{
+		context: scannerContext{
+			resource: s.mockResource,
+			cfg:      Config{},
+		},
+		tallyScope: s.mockScope,
+		zapLogger:  zap.NewNop(),
+		startWorkflowWithRetryFn: func(_ string, _ time.Duration, _ commonResource.Resource, _ func(client client.Client) error) error {
+			return nil
+		},
+		newWorkerFn: func(_ workflowserviceclient.Interface, _ string, _ string, _ worker.Options) worker.Worker {
+			return s.mockWorker
+		},
+	}
 }
 
 func (s *scannerTestSuite) TearDownTest() {
 	s.mockCtrl.Finish()
 }
+
+func (s *scannerTestSuite) TestNew() {
+	params := &BootstrapParams{
+		Config:     Config{},
+		TallyScope: s.mockScope,
+	}
+
+	s.NotNil(New(s.mockResource, params))
+}
+
+func (s *scannerTestSuite) TestStart() {
+	testCases := []struct {
+		name       string
+		cfg        Config
+		setupMocks func()
+		err        error
+	}{
+		{
+			name: "with shard scanners",
+			cfg: Config{
+				ShardScanners: []*shardscanner.ScannerConfig{
+					{
+						ScannerWFTypeName: "testScannerWF1",
+						DynamicParams: shardscanner.DynamicParams{
+							ScannerEnabled: func(opts ...dynamicconfig.FilterOption) bool {
+								return true
+							},
+							FixerEnabled: func(opts ...dynamicconfig.FilterOption) bool {
+								return true
+							},
+						},
+						StartWorkflowOptions: client.StartWorkflowOptions{
+							TaskList: "shard-scanner-task-list-1",
+						},
+						ScannerHooks: func() *shardscanner.ScannerHooks {
+							return &shardscanner.ScannerHooks{
+								GetScannerConfig: func(scanner shardscanner.ScannerContext) shardscanner.CustomScannerConfig {
+									return map[string]string{}
+								},
+							}
+						},
+						FixerHooks: func() *shardscanner.FixerHooks {
+							return &shardscanner.FixerHooks{
+								GetFixerConfig: func(fixer shardscanner.FixerContext) shardscanner.CustomScannerConfig {
+									return map[string]string{}
+								},
+							}
+						},
+					},
+				},
+				Persistence: &config.Persistence{
+					DefaultStore: "nosql",
+					DataStores: map[string]config.DataStore{
+						"nosql": {
+							NoSQL: &config.NoSQL{},
+						},
+					},
+				},
+				HistoryScannerEnabled: func(opts ...dynamicconfig.FilterOption) bool {
+					return false
+				},
+			},
+			setupMocks: func() {
+				s.mockWorker.EXPECT().Start().Return(nil).Times(2)
```

Comment:
maybe worth a comment in code here:
it's a test-suite quirk that this is the same `mockWorker` called 2x.  normally there would be two separate workers, each started once.

which seems fine for these tests, it's just misleading to look at.